### PR TITLE
Add nav service updates

### PR DESCRIPTION
**Pull request by holtzmak**
In this PR I have:
1. Added test(s) for `NavService`
2. Made things that use the Main Navigator use `NavService`
3. Identified that we have to use `Navigator.of(context)` when using inner navigators because they don't have a service
4. Removed the `Drawer` from useless places (like `WelcomeScreen`)
5. Fixed broken tests from changes

There are no GUI changes in this PR (besides the removal of excess drawers in places they shouldn't have been).

@MummenRider I've noticed that the navigation `Drawer` is not persistent in all pages after `HomeScreen`. This confuses me, because the bottom nav bar is persistent and constructed at the same time as the drawer in `HomeScreen`. Any ideas?

Also, @MummenRider what do you think about adding services for the inner navigators? The current service is just the main navigator.

### CODE REVIEW

File: app/lib/core/services/navigation/nav_service.dart
Comment (MummenRider) [inline app/lib/core/services/navigation/nav_service.dart:18]: Wrapping arguments with curly braces means it's optional already, I'm pretty sure if-else isn't necessary here. This applies to all nav services methods you did.

File: app/lib/ui/views/new/new_screen.dart
Comment (MummenRider) [inline app/lib/ui/views/new/new_screen.dart]: You're not supposed to use nav service in UI layer. Here's what you need to do:
```
class NewScreenViewModel extends BaseViewModel{
final NavigationService _navigationService = locator<NavigationService>();
void navigateToTestScreenThree() => _navigationService.navigateTo(TestScreenThree.route);
}
```
In your view, you can simply do: `model.navigateToTestScreenThree();`
Note: This apply to every screen that will navigate

**Review by MummenRider — CHANGES_REQUESTED** [review]
LGTM! Made suggestion when using navigation service . I'm pretty sure you can use NavigatorService in inner loop too. For each inner screen that requires navigation, you simply create a viewmodel for it. I made suggestion that shows how  to implement it

File: app/lib/core/services/navigation/nav_service.dart
Comment (holtzmak) [reply app/lib/core/services/navigation/nav_service.dart:18]: I think I sent you this before:
```
The following NoSuchMethodError was thrown while handling a gesture:
The method 'pushNamed' was called on null.
Receiver: null
Tried calling: pushNamed<Object>("/A", arguments: null)
```
We still need to handle null case, hence why I cast to optional as an easy way to check without forcing people to add `Optional.empty()` when using this function.

Comment (MummenRider) [reply app/lib/core/services/navigation/nav_service.dart:18]: oh okay, I didn't know the full context of that error message. If this fixes the problem, then LGTM!

File: app/lib/ui/views/welcome/welcome_screen.dart
Comment (MummenRider) [inline app/lib/ui/views/welcome/welcome_screen.dart:27]: I think you forgot to add `()` at the end?

**Review by MummenRider — APPROVED** [review]
I'll approve this, but please change it before merging. Thanks!

File: app/lib/ui/views/welcome/welcome_screen.dart
Comment (holtzmak) [reply app/lib/ui/views/welcome/welcome_screen.dart:27]: I did not! This is an uncommon way of specifying the function to call.

This is the same as `() => model.navigateToHomeScreen()`
> So, rather than make an anonymous function to call a function, I just say call this function `model.navigateToHomeScreen`. 

Functions can be pointed to like variables, adding `()` calls it, and `() => {}` declares a function. Since `navigateToHomeScreen` is already declared, I just reference it here.

**Pull request merged**
